mlflow/server/js/src/experiment-tracking/components/ExperimentRunsTableMultiColumnView2.js

```diff
@@ -475,8 +475,10 @@ export class ExperimentRunsTableMultiColumnView2 extends React.Component {
       prevProps.metricKeyList.length !== this.props.metricKeyList.length ||
       prevProps.paramKeyList.length !== this.props.paramKeyList.length ||
       prevProps.visibleTagKeyList.length !== this.props.visibleTagKeyList.length ||
-      prevProps.categorizedUncheckedKeys[COLUMN_TYPES.ATTRIBUTES].length !==
-        this.props.categorizedUncheckedKeys[COLUMN_TYPES.ATTRIBUTES].length ||
```

Comment:
@jimmyxu-db This line was the root cause of the column selector issue. 

When the checkbox (for the source column) is selected, `prevProps.categorizedUncheckedKeys[COLUMN_TYPES.ATTRIBUTES]` returns `['Source']`. When it's unselected, `[null]` is returned. I think the code block below causes this behavior. This means toggling the checkbox doesn't change the length of `prevProps.categorizedUncheckedKeys[COLUMN_TYPES.ATTRIBUTES]`, which prevents `this.setColumnDefs()` (at line 484) from being called.

https://github.com/mlflow/mlflow/blob/3ba3c3f8119d68d617b680241bf010f5015979da/mlflow/server/js/src/experiment-tracking/components/ExperimentPage.js#L383-L399